server/src/main/java/com/datastrato/gravitino/server/web/rest/PartitionOperations.java

```diff
@@ -102,14 +102,14 @@ public Response getPartition(
           httpRequest,
           () -> {
             NameIdentifier tableIdent = NameIdentifier.of(metalake, catalog, schema, table);
-            return TreeLockUtils.doWithTreeLock(
-                tableIdent,
-                LockType.READ,
-                () -> {
-                  Table loadTable = dispatcher.loadTable(tableIdent);
-                  Partition p = loadTable.supportPartitions().getPartition(partition);
-                  return Utils.ok(new PartitionResponse(DTOConverters.toDTO(p)));
-                });
+            Table loadTable = dispatcher.loadTable(tableIdent);
+
+            Partition p =
```

Comment:
As we have moved the logic about tree lock into `dispatcher`, for partitioning-related logic, we can't completely remove tree lock from APIs.

This seems to be not very elegant, but I can't find a better one till now.

Reply:
The change here will possibly lead to inconsistency, right?

Reply:
Yeah, I'm working on it.

Reply:
In case of this, I would rethink the necessity of this change.

Reply:
OK, I have some doubts about whether PR is necessary.

Reply:
You can defer this and work on others firstly.